profiler/controllers.py

```diff
@@ -1,70 +1,175 @@
-        # can't close the stream even in a context manager: it'll be needed
-        # after the return from this method, we'll let Python's GC do its job
-        return request.make_response(
-            stream,
-            headers=[
-                ('Content-Disposition',
-                 'attachment; filename="%s"' % filename),
-                ('Content-Type', 'application/octet-stream')
-            ], cookies={'fileToken': token})
-
-    @openerpweb.jsonrequest
-    def initial_state(self, request):
-        user = request.session.model('res.users')
+        exclude_fname = self.get_exclude_fname()
+        with tools.osutil.tempdir() as dump_dir:
+            ts = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+            filename = 'openerp_%s' % ts
+            stats_path = os.path.join(dump_dir, '%s.stats' % filename)
+            core.profile.dump_stats(stats_path)
+            _logger.info("Pstats Command:")
+            params = {'fnames': stats_path, 'sort': 'cumulative', 'limit': 45,
+                      'exclude_fnames': exclude_fname}
+            _logger.info(
+                "fnames=%(fnames)s, sort=%(sort)s,"
+                " limit=%(limit)s, exclude_fnames=%(exclude_fnames)s", params)
+            pstats_list = get_pstats_print2list(**params)
+            with Capturing() as output:
+                print_pstats_list(pstats_list)
+            result_path = os.path.join(dump_dir, '%s.txt' % filename)
+            with open(result_path, "a") as res_file:
+                for line in output:
+                    res_file.write('%s\n' % line)
+            # PG_BADGER
+            self.dump_pgbadger(dump_dir, 'pgbadger_output.txt')
+            t_zip = tempfile.TemporaryFile()
+            tools.osutil.zip_dir(dump_dir, t_zip, include_dir=False)
+            t_zip.seek(0)
+            headers = [
+                ('Content-Type', 'application/octet-stream; charset=binary'),
+                ('Content-Disposition', content_disposition(
+                    '%s.zip' % filename))]
+            _logger.info('Download Profiler zip: %s', t_zip.name)
+            return request.make_response(
+                t_zip, headers=headers, cookies={'fileToken': token})
+
+    @http.route(['/web/profiler/initial_state'], type='json', auth="user")
+    def initial_state(self, **post):
+        user = request.env['res.users'].browse(request.uid)
         return {
-            'has_player_group': user.has_group('profiler.group_profiler_player'),
+            'has_player_group': user.has_group(
+                'profiler.group_profiler_player'),
             'player_state': ProfilerController.player_state,
         }
+
+    def dump_pgbadger(self, dir_dump, output):
+        pgbadger = find_in_path("pgbadger")
+        if not pgbadger:
+            raise Exception("pgbadger not found")
+        filename = os.path.join(dir_dump, output)
+        logfilename = os.path.join(dir_dump, 'postgresql.log')
```

Comment:
I think it is, in order to have the possibility of run pgbadger with  a different set of options